pkg/certificate/manager.go

```diff
@@ -21,11 +20,12 @@ var (
 )
 
 // NewManager creates a new CertificateManager with the passed MRCClient and options
-func NewManager(ctx context.Context, mrcClient MRCClient, getServiceCertValidityPeriod func() time.Duration, getIngressCertValidityDuration func() time.Duration, checkInterval time.Duration) (*Manager, error) {
+func NewManager(ctx context.Context, mrcClient MRCClient, getServiceCertValidityPeriod func() time.Duration, getIngressCertValidityDuration func() time.Duration, checkInterval time.Duration, ownedCertTypes []CertificateType) (*Manager, error) {
```

Comment:
isOwnedCertTypes an all or nothing thing? My guess is yet. What if we simplified for now, and just had a bool for enabling the MRCReconciler

Reply:
No, ownedCertTypes is how the top-level control plane command is able to claim ownership over one or more types of certificates (e.g sidecar, Gateway, etc). This is so  the cert manager knows which certificate status fields on the MRC to update

Reply:
but is there ever a use case where a control plane component uses some of these but not all?

Reply:
Yes, the sets are disjoint. The injector only handles the mutating webhook and bootstrap certs, and the controller owns the rest